sunpy/io/_asdf.py

```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+from collections import OrderedDict
+
+import numpy as np
+
+import asdf
+
+from sunpy.io._header import FileHeader
+
+__all__ = ["write", "read", "get_header", "get_keys_name"]
+
+def write(fname, data, header, **kwargs):
+    """
+    Take ``(data, header)`` pairs and save it to an ASDF file.
+
+    Parameters
+    ----------
+    fname : `str`
+        File name, with extension.
+    data : `numpy.ndarray`
+        N-dimensional data array.
+    header : `dict`
+        A header dictionary.
+    """
+    map_name = Path(fname)
+    map_name = map_name.name
+    ty = data.dtype
+    data = data.astype(ty)
+    meta = dict(header)
+    asdf.AsdfFile({map_name : {"meta" : meta, "data" : data}},lazy_load=False,**kwargs).write_to(fname)
+
+
+def read(fname, **kwargs):
+    """
+    Read an ASDF file.
+
+    Parameters
+    ----------
+    filepath : `str`
+        The ASDF file to be read.
+
+    Returns
+    -------
+    `list`
+        A list of ``(data, header)`` tuples.
+    """
+
+    with asdf.open(fname) as af:
+        map_name = get_keys_name(fname)
+        if isinstance(af[map_name],dict):
```

Comment:
```suggestion
        if isinstance(af[map_name], dict):
```
Can you try to make sure the whitespace is correct for python code please?